### PR TITLE
More dynamic state refactoring

### PR DESCRIPTION
**Pull request by spencer-lunarg**
More improvements to make the Dynamic State code cleaner

For some reason the `git` diff looks way worse then it actually is for `core_validation.cpp`

### CODE REVIEW

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -1967,1464 +1963,1465 @@ bool CoreChecks::ValidatePipelineLibraryFlags(const VkGraphicsPipelineLibraryFla
...
-                         "vkCreateGraphicsPipelines() pCreateInfos[%" PRIu32
-                         "]: Extended dynamic state used by the "
-                         "extendedDynamicState3ShadingRateImageEnable "
-                         "feature is not enabled",
-                         pipe_index);
-    }
+    skip |= ValidatePipelineCacheControlFlags(pipeline->GetPipelineCreateFlags(), pipe_index, "vkCreateGraphicsPipelines",
+                                              "VUID-VkGraphicsPipelineCreateInfo-pipelineCreationCacheControl-02878");
+    skip |= ValidatePipelineProtectedAccessFlags(pipeline->GetPipelineCreateFlags(), pipe_index);
+    skip |= ValidateGraphicsPipelineDynamicStateFeature(pipeline, pipe_index);
```
Comment (spencer-lunarg) [inline layers/core_validation.cpp:3424]: all I did was move everything here into a new `ValidateGraphicsPipelineDynamicStateFeature` and no matter where I put it in this file, `git` is acting like the change is much worse then it looks

**Review by pdaniell-nv — APPROVED** [review]
I think it's okay. The core_validation.cpp changes were impossible to review so I have to trust it was just a simple move.

**Comment by spencer-lunarg** [comment]
I decided to just remove the core validation change, I am not a fan of what ever `git` was doing, seems to be a potential `git` bug but not worth fighting here, will that part refactor later

File: layers/parameter_validation_utils.cpp [hunk 2]
```diff
@@ -2723,6 +2017,18 @@ bool StatelessValidation::manual_PreCallValidateCreateGraphicsPipelines(VkDevice
                                  i, feedback_struct->pipelineStageCreationFeedbackCount, create_info.stageCount);
             }
 
+            // helpers for bool used multiple times below
+            const bool has_dynamic_viewport = dynamic_state_map.find(VK_DYNAMIC_STATE_VIEWPORT) != dynamic_state_map.end();
```
[the remark this answers is not in the document]
Comment (spencer-lunarg) [reply layers/parameter_validation_utils.cpp:2021]: + :100: - didn't see the `Contains` function, was only looking at std spec

File: layers/parameter_validation_utils.cpp [hunk 3]
```diff
@@ -1819,72 +1819,6 @@ bool StatelessValidation::manual_PreCallValidateCreateGraphicsPipelines(VkDevice
 
     if (pCreateInfos != nullptr) {
         for (uint32_t i = 0; i < createInfoCount; ++i) {
-            bool has_dynamic_viewport = false;
```
[the remark this answers is not in the document]
Comment (spencer-lunarg) [reply layers/parameter_validation_utils.cpp:1822]: they are now the `unordered_map`, most of these were for was the "is the state duplicated" check

**Pull request merged**